waspc/src/Wasp/JsImport.hs

```diff
@@ -0,0 +1,80 @@
+{-# LANGUAGE DeriveDataTypeable #-}
+
+module Wasp.JsImport
+  ( JsImport (..),
+    JsImportName (..),
+    JsImportData,
+    JsImportAlias,
+    JsImportPath,
+    getJsImportData,
+  )
+where
+
+import Data.Data (Data)
+import Data.List (isPrefixOf)
+import StrongPath (Dir', File', Path, Posix, Rel)
+import qualified StrongPath as SP
+
+data JsImport = JsImport
+  { -- | What is being imported.
+    _name :: JsImportName,
+    -- | Path from which we are importing.
+    _path :: JsImportPath,
+    -- | Alias for the imported name
+    _importAlias :: Maybe JsImportAlias
+  }
+  deriving (Show, Eq, Data)
+
+type JsImportPath = Path Posix (Rel Dir') File'
+
+type Identifier = String
+
+type JsImportAlias = String
+
+data JsImportName
+  = -- | Represents external imports like @import Identifier from "file.js"@
+    JsImportModule Identifier
+  | -- | Represents external imports like @import { Identifier } from "file.js"@
+    JsImportField Identifier
+  deriving (Show, Eq, Data)
+
+type JsImportIdentifier = String
+
+type JsImportWhat = String
+
+type JsImportStatement = String
+
+type JsImportData = (JsImportIdentifier, JsImportStatement)
```

Comment:
Ha this name is really hm. In general, names with `data`, `info` and similar in them are a bad pattern, since they don't really mean anything.
In this case, a better name would probably be `JsImportIdentifierAndStmt`. Or maybe even better to avoid having a `type` here if we are having hard time finding a name for it.

Reply:
I've gone with using the tuple everywhere explictly :)